jwst/resample/resample_spec.py

```diff
@@ -91,73 +92,155 @@ def build_interpolated_output_wcs(self, refmodel=None):
-            x_tan_array = x_tan[lam_center_index]
-            y_tan_array = y_tan[lam_center_index]
-        x_tan_array = x_tan_array[~np.isnan(x_tan_array)]
-        y_tan_array = y_tan_array[~np.isnan(y_tan_array)]
-
-        fitter = LinearLSQFitter()
-        fit_model = Linear1D()
-        pix_to_ra = fitter(fit_model, np.arange(x_tan_array.shape[0]), x_tan_array)
-        pix_to_dec = fitter(fit_model, np.arange(y_tan_array.shape[0]), y_tan_array)
-
+        # for each input model convert slit x,y to ra,dec,lam
+        # use first input model to set spatial scale
+        # use center of appended ra and dec arrays to set up
+        # center of final ra,dec
+        # append all ra,dec, wavelength array for each slit
+        # use first model to initialize wavelenth array
+        # append wavelengths that fall outside the endpoint of
+        # of wavelength array when looping over additional data
+
+        all_wavelength = []
+        all_ra_slit = []
+        all_dec_slit = []
+
+        for im, model in enumerate(self.input_models):
+            wcs = model.meta.wcs
+            bb = wcs.bounding_box
+            grid = wcstools.grid_from_bounding_box(bb)
+            ra, dec, lam = np.array(wcs(*grid))
+            spectral_axis = find_dispersion_axis(model)
+            spatial_axis = spectral_axis ^ 1
+
+            # Compute the wavelength array, trimming NaNs from the ends
+            wavelength_array = np.nanmedian(lam, axis=spectral_axis)
+            wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
+
+            # need to estimate the spatial sampling to use for the output WCS
+            # it is assumed the spatial sampling is the same for all the input
+            # models. So we can use the first input model to set the spatial
+            # sampling.
+            # Note: using the first input model sets up the spatial output to match
+            # this  model. For nodded (background subtracted data)
+            # this means the positive and negative traces will have the same
+            # configurations as the first input model.
+
+            # Steps to do this for first input model:
+            # 1. find the center of slit in wavelength
+            # 2. Pull out the ra and dec at the center of the slit.
+            # 3. Find the mean ra,dec and the center of the slit this will
+            #    represent the tangent point
+            # 4. Convert ra,dec -> tangent plane projection: x_tan,y_tan
+            # 5. using x_tan, y_tan perform a linear fit to find spatial sampling
+            # first input model sets intializes wavelength array and defines
+            # the spatial scale of the output wcs
+            if im == 0:
+                for iw in wavelength_array:
+                    all_wavelength.append(iw)
+
+                lam_center_index = int((bb[spectral_axis][1] -
+                                            bb[spectral_axis][0]) / 2)
+                if not spatial_axis:
```

Comment:
I know this syntax works mathematically and programmatically, but to a human reading it the immediate interpretation is "we're working with an axis that isn't spatial". I think it would be more obvious for readability to just use "if spatial_axis == 0"